pkg/apis/storage/validation/validation.go

```diff
@@ -266,3 +266,178 @@ func validateAllowedTopologies(topologies []api.TopologySelectorTerm, fldPath *f
 
 	return allErrs
 }
+
+// ValidateCSINodeInfo validates a CSINodeInfo.
```

Comment:
We introduced a version of `validateCSINodeInfo` a while back at https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/nodeinfomanager/nodeinfomanager.go#L618 that performs some additional validations of the fields of CSI NodeInfo. Should the `ValidateCSINodeInfo` here implement the same in-depth validation and potentially be the only version in k/k? 

The logic above was a temporary workaround to deal with the lack of a universal way to deploy a webhook to perform this validation for instances of the node info CRD.

Reply:
Does the status field always have to be populated at the same time as the spec field?  Could the spec be populated first without the status?

Reply:
I believe the set of drivers in spec and status fields needs to be in sync based on this unit test `mismatch between drivers in specs and status (specs subset of status)`: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/nodeinfomanager/nodeinfomanager_test.go#L847

Reply:
right now they are populated at the "same time", but in the future when we move to a reconciler model (which there are plans to do) the spec may be populated before the status is determined. We need to make sure that whatever we do now is either compatible with that new model, or can be easily changed to be compatible to that paradigm.

Reply:
Added checks for duplicate entries in specs and statuses.  I think the check for matching drivers in specs and statuses can stay in nodeinfomanager for now?

Reply:
I don't think you should rely on them being in sync (or if so we can maybe flatten it all to spec).  Most spec/status resources only allow to write to one or the other but not both at a give API call.  You do need to spec what happens if they fall out of sync.